daemon/daemon.go

```diff
@@ -359,7 +360,21 @@ func (daemon *Daemon) restore() error {
 			if daemon.configStore.AutoRestart && container.shouldRestart() {
 				logrus.Debugf("Starting container %s", container.ID)
 
-				if err := daemon.containerStart(container); err != nil {
+				var retries int
+				for err = daemon.containerStart(container); err != nil && retries <= 5; retries++ {
+					// Best effort to wait for container deps to be up
+					switch e := err.(type) {
+					case errcode.Error:
+						if e.ErrorCode() == derr.ErrorCodeLinkNotRunning {
```

Comment:
@duglin ?

Also, I have the switch because I was also trying to fix another case where overlay networks aren't ready -- but there appears to be a deeper issue in libnetwork that needs to get fixed first.